app/controllers/application_controller.rb

```diff
@@ -634,6 +615,10 @@ def build_ae_tree(type = :ae, name = :ae_tree)
     tree
   end
 
+  def filename_timestamp(basename, format = 'fname')
```

Comment:
You are right. This is not a natural place for the method.

However, I am afraid that making this helper will only expose it to views. Not much more. Perhaps we can put it into `lib/`? However, then it will lose the context, I haven't found good (existing) file to fit this. Still, I am quite new to this rails thing, so I could get this whole observation wrong, why does helper came to your mind?

Perhaps it makes sense to kick all the methods concerned with report creation/rendering to a concern? I am volunteering to give it try.

Thanks!